iexec-sms-library/src/main/java/com/iexec/sms/api/SmsService.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.sms.api;
+
+import com.iexec.common.chain.WorkerpoolAuthorization;
+import com.iexec.common.tee.TeeWorkflowSharedConfiguration;
+import com.iexec.common.utils.BytesUtils;
+import feign.FeignException;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Optional;
+
+@Slf4j
+public class SmsService {
+
+    private final SmsClient smsClient;
+
+    private TeeWorkflowSharedConfiguration teeWorkflowConfiguration;
+
+    public SmsService(SmsClient smsClient) {
+        this.smsClient = smsClient;
+    }
+
+    public boolean addAppDeveloperAppComputeSecret(String authorization,
+                                                   String appAddress,
+                                                   //String secretIndex,
+                                                   String secretValue) {
+        try {
+            smsClient.addAppDeveloperAppComputeSecret(authorization, appAddress, secretValue);
+            return true;
+        } catch (FeignException e) {
+            log.error("addAppDeveloperAppComputeSecret failed", e);
+        }
+        return false;
+    }
+
+    public boolean isAppDeveloperAppComputeSecretPresent(String appAddress, String secretIndex) {
+        try {
+            smsClient.isAppDeveloperAppComputeSecretPresent(appAddress, secretIndex);
+            return true;
+        } catch(FeignException e) {
+            log.error("isAppDeveloperAppComputeSecretPresent failed", e);
+        }
+        return false;
+    }
+
+    public Integer getMaxRequesterSecretCountForAppCompute(String appAddress) {
+        try  {
+            return smsClient.getMaxRequesterSecretCountForAppCompute(appAddress).getData();
```

Comment:
Should not happen, but might be better to handle NPE?

Reply:
How would you handle/implement this ?

On server side, the method `getMaxRequesterSecretCountForAppCompute` always return an `ApiResponseBody` instance.
The `data` field of this instance could be `null`, which is already the value returned if the REST call fails or returns a non 2xx status code.

The library consumer needs to handle the case where the returned value is `null` or we could throw an exception.
But the consumer would still need to react and handle the error depending on its own busines features/requirements.